### PR TITLE
Jsonschema

### LINKED ISSUE #22: Read: the use of aggregate may be insecure
Aggregation can lead to a lot of shooting oneself in the foot. This should not be accessible through the API as it is a bit too advanced.

At the very least, the allowed aggregations should be validated to ensure the integrity of the database.

### PR DESCRIPTION
**Pull request by hemangandhi**
Uses `jsonschema` instead of if statements for the validation of inputs.

Also adds a few utilities and then makes them widespread.

There is some cute continuation passing style and readability and testing coverage go up.

This should resolve #22 and #30 fully.

### CODE REVIEW

File: authorize.py [hunk 1]
```diff
@@ -70,19 +75,35 @@ def authorize_then_consume(event, context):
             rv['statusCode'] = 400
     return rv
 
-def create_user(event, context, mlh = False):
+@ensure_schema({
```
Comment (Sail338) [inline authorize.py]: What if people dont fill in anything  for those fields

File: authorize.py [hunk 2]
```diff
@@ -70,19 +75,35 @@ def authorize_then_consume(event, context):
...
+        "major": {"type": "string"},
+        "short_answer": {"type": "string"},
+        "shirt_size": {"type": "string"},
+        "first_name": {"type": "string"},
+        "last_name": {"type": "string"},
+        "dietary_restrictions": {"type": "string"},
+        "special_needs": {"type": "string"},
+        "date_of_birth": {"type": "string"},
+        "school": {"type": "string"},
+        "grad_year": {"type": "string"},
```
Comment (Sail338) [inline authorize.py]: this should prob be an in

File: authorize.py [hunk 3]
```diff
@@ -70,19 +75,35 @@ def authorize_then_consume(event, context):
...
+        "gender": {"type": "string"},
+        "level_of_study": {"type": "string"},
+        "ethnicity": {"type": "string"},
+        "phone_number": {"type": "string"}
```
Comment (Sail338) [inline authorize.py]: Might be worth making this an int

File: read.py [hunk 4]
```diff
@@ -1,67 +1,93 @@
...
-def validate_user(db, token, email):
-    """
-    Returns the user if the authtoken provided as token is valid.
-    """
-    if not token or not email:
-        return False
+def tidy_results(res):
+    for i in res:
+        del i['_id']
+        del i['password']
```
Comment (Sail338) [inline read.py]: me like

Comment (hemangandhi) [reply read.py]: This is for JSON serialization - the two fields are some sort of raw byte streams that AWS doesn't serialize if we returned them.

File: authorize.py (hunk 3 above)
Comment (hemangandhi) [reply authorize.py]: Will make an issue so that frontend can catch up after this hackru

File: read.py (hunk 4 above)
Comment (Sail338) [reply read.py]: oh got it

File: authorize.py (hunk 2 above)
Comment (hemangandhi) [reply authorize.py]: Yep. When that happens, I'll make it automatically restrict to current seniors onwards.

It's a string to not mess with the frontend.

Comment (Sail338) [reply authorize.py]: I see, ok we can make issues on front to change for next hackru

**Pull request merged**